src/os/bluestore/BlueStore.cc

```diff
@@ -6289,6 +6289,8 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       txc->log_state_latency(logger, l_bluestore_state_prepare_lat);
       if (txc->ioc.has_pending_aios()) {
 	txc->state = TransContext::STATE_AIO_WAIT;
+	++txc->osr->txc_with_unstable_io;
+	txc->had_ios = true;
```

Comment:
I think ++txc->osr->txc_with_unstable_io can be moved right after `case TransContext::STATE_IO_DONE:` which is executed with holding osr->qlock.
Like `case TransContext::STATE_IO_DONE: if (txc->has_ios) { ++txc->osr->txc_with_unstable_io }

Otherwise, a recent transaction may prevent KernelDevice.aio_thread from doing RTC with earlier transaction